martian-filetypes/src/lib.rs

```diff
@@ -227,6 +227,20 @@ pub trait FileTypeWrite<T>: MartianFileType {
     fn write_into<W: io::Write>(writer: W, item: &T) -> Result<(), Error>;
 }
 
+/// Something that can create a Martian file and eagerly write into it.
+pub trait FileTypeCreate<T, F: FileTypeWrite<T>> {
+    /// Create a Martian file of type F and immediately write T into it.
+    fn create_file(&self, file_name: &str, contents: &T) -> Result<F, Error>;
+}
+
+impl<T, F: FileTypeWrite<T>> FileTypeCreate<T, F> for MartianRover {
+    fn create_file(&self, file_name: &str, contents: &T) -> Result<F, Error> {
+        let file: F = self.make_path(file_name);
+        file.write(contents)?;
+        Ok(file)
+    }
+}
```

Comment:
Would it be better to add `with_content(self, item: &T) -> Result<Self, Error>` to `FileTypeWrite`?

Then we can write something like:
```
rover.make_path(filename).with_content(content)?;
```

Reply:
Ooh an interesting take, method chaining would be nice here.  I'll give that a try!

Reply:
Yes, much nicer!